models/evc.py

```diff
@@ -810,41 +813,50 @@ def get_failover_flows(self):
 
     def _prepare_direct_uni_flows(self):
         """Prepare flows connecting two UNIs for intra-switch EVC."""
-        vlan_a = self.uni_a.user_tag.value if self.uni_a.user_tag else None
-        vlan_z = self.uni_z.user_tag.value if self.uni_z.user_tag else None
+        vlan_a = self._get_value_from_uni_tag(self.uni_a)
+        vlan_z = self._get_value_from_uni_tag(self.uni_z)
 
-        is_EVPL = (vlan_a is not None)
         flow_mod_az = self._prepare_flow_mod(
             self.uni_a.interface, self.uni_z.interface,
-            self.queue_id, is_EVPL
+            self.queue_id, vlan_a
         )
-        is_EVPL = (vlan_z is not None)
         flow_mod_za = self._prepare_flow_mod(
             self.uni_z.interface, self.uni_a.interface,
-            self.queue_id, is_EVPL
+            self.queue_id, vlan_z
         )
 
-        if vlan_a and vlan_z:
+        if vlan_a is not None:
             flow_mod_az["match"]["dl_vlan"] = vlan_a
+
+        if vlan_z is not None:
             flow_mod_za["match"]["dl_vlan"] = vlan_z
+
+        if vlan_z not in {None, "4096/4096", 0}:
```

Comment:
Reading the PR again, I noticed that this tuple `{None, "4096/4096", 0}` appears six times (counting on tests too), it'll likely not change, so probably won't result in potential refactoring typo issues in the future, but if you manage to replace this as a constant with a name that's meaningful for the developer reading it that'd be even better.

Reply:
Looks good. Let me close this thread.